### PR TITLE
Feature/distance sensor

### PR DESCRIPTION
**Pull request by LandryNorris**
Create 2m distance sensor module and sample.

### CODE REVIEW

**Comment by LandryNorris** [comment]
The code matches the reference as close as reasonable. We may want to remove some extra methods we don't use from the reference.

File: packages/distance-sensor/src/distance-sensor.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+    async setup() {
+        await this.device.setup();
+    }
+
+    async getDistanceMillimeters(): Promise<number> {
+        return await this.device.getDistanceMillimeters();
+    }
+
+    startMeasuringDistance(onDistanceRecorded: (_: number) => void,
```
Comment (NoahAndrews) [inline packages/distance-sensor/src/distance-sensor.ts:20]: The parameter passed to `onDistanceRecorded()` should have a name that indicates the unit.

File: packages/sample/src/commands/distance.ts
Comment (NoahAndrews) [inline packages/sample/src/commands/distance.ts:1]: By default, the tool should provide a one-shot value, but have a `--continuous` option that makes it keep outputting values indefinitely (not just for 20 seconds).

File: packages/distance-sensor/src/distance-sensor.ts
Comment (NoahAndrews) [inline packages/distance-sensor/src/distance-sensor.ts:1]: The functions in this file need to be documentd.

**Review by NoahAndrews — CHANGES_REQUESTED** [review]

File: packages/sample/src/commands/distance.ts
Comment (LandryNorris) [reply packages/sample/src/commands/distance.ts:1]: Added continuous flag

File: packages/distance-sensor/src/distance-sensor.ts (hunk 1 above)
Comment (LandryNorris) [reply packages/distance-sensor/src/distance-sensor.ts]: Added parameter name and mentioned units in documentation

File: packages/distance-sensor/package.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
+{
+  "name": "@rev-robotics/distance-sensor",
+  "version": "0.1.0",
+  "description": "Library for using the REV Robotics 2m distance sensor",
+  "main": "dist/index.js",
+  "type": "module",
+  "dependencies": {
+    "serialport": "^10.5.0",
+    "@rev-robotics/expansion-hub": "*"
```
Comment (NoahAndrews) [inline packages/distance-sensor/package.json:9]: This needs a more appropriate value.

File: packages/distance-sensor/src/i2c-utils.ts [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+export async function readRegisterMultipleBytes(
+    hub: ExpansionHub,
+    channel: number,
+    address: number,
+    register: number,
+    n: number
+): Promise<number[]> {
+    await hub.writeI2CSingleByte(channel, address, register);
+    await hub.readI2CMultipleBytes(channel, address, n);
```
Comment (NoahAndrews) [inline packages/distance-sensor/src/i2c-utils.ts:23]: `ExpansionHub.writeI2CReadMultipleBytes()` exists for this (though that's a terrible name, and its `startAddress` parameter name is also very unclear).

**Review by NoahAndrews — DISMISSED** [review]

File: packages/core/src/ExpansionHub.ts [hunk 4]
```diff
@@ -112,34 +112,92 @@ export interface ExpansionHub extends RevHub {
     getDigitalAllInputs(): Promise<number>;
 
     // I2C
+    /**
+     * Configure the speed for a I2C channel
```
Comment (NoahAndrews) [inline packages/core/src/ExpansionHub.ts:116]:
```suggestion
     * Configure the speed of an I2C channel
```

File: packages/core/src/ExpansionHub.ts [hunk 5]
```diff
@@ -112,34 +112,92 @@ export interface ExpansionHub extends RevHub {
...
+     * @param i2cChannel
+     * @param speedCode
+     */
     setI2CChannelConfiguration(
         i2cChannel: number,
         speedCode: I2CSpeedCode,
     ): Promise<void>;
+
+    /**
+     * Get the speed for a I2C channel
```
Comment (NoahAndrews) [inline packages/core/src/ExpansionHub.ts:126]:
```suggestion
     * Get the speed of an I2C channel
```

**Review by NoahAndrews — CHANGES_REQUESTED** [review]

**Review by NoahAndrews — APPROVED** [review]

**Pull request merged**